### PR TITLE
Do not capture Ignored characters by default

### PR DESCRIPTION
**Pull request by bbakerman**
See #2207

This will save memory by default.  graphql-java will not capture ignored characters by default.

A static is provided to allow a JVM wide default (set to false) or new parse methods will allow it to be explicitly set
on the parse method call

### CODE REVIEW

File: src/main/java/graphql/parser/Parser.java [hunk 1]
```diff
@@ -133,8 +176,24 @@ public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int
      * @param multiSourceReader the source of the query document
      *
      * @return a new GraphqlAntlrToLanguage instance
+     *
+     * @deprecated - really should use {@link #getAntlrToLanguage(CommonTokenStream, MultiSourceReader, Boolean)}
      */
+    @Deprecated
     protected GraphqlAntlrToLanguage getAntlrToLanguage(CommonTokenStream tokens, MultiSourceReader multiSourceReader) {
```
Comment (bbakerman) [inline src/main/java/graphql/parser/Parser.java:183]: This was left to preserve old behavior - if some one overrides this then they can.

File: src/test/groovy/graphql/parser/ParserTest.groovy [hunk 2]
```diff
@@ -69,11 +71,11 @@ class ParserTest extends Specification {
```
Comment (bbakerman) [inline src/test/groovy/graphql/parser/ParserTest.groovy:72]: This class was littered with IDEA warnings.  I clean up a lot because broken windows and all that

File: src/test/groovy/graphql/parser/ParserTest.groovy [hunk 3]
```diff
@@ -842,6 +844,26 @@ triple3 : """edge cases \\""" "" " \\"" \\" edge cases"""
         document.getAdditionalData().get("key") == "value"
         document.children[0].getAdditionalData().get("key") == "value"
```
Comment (bbakerman) [inline src/test/groovy/graphql/parser/ParserTest.groovy:846]: the old method is tested here but the diff does not show it

File: src/test/groovy/graphql/parser/ParserTest.groovy [hunk 4]
```diff
@@ -727,7 +729,7 @@ triple3 : """edge cases \\""" "" " \\"" \\" edge cases"""
         def input = "{,\r me\n\t} ,\n"
 
         when:
-        Document document = new Parser().parseDocument(input)
+        Document document = new Parser().parseDocument(input, true)
```
Comment (bbakerman) [inline src/test/groovy/graphql/parser/ParserTest.groovy:732]: Again the diff does not show it but this is the old ignored chars test - now needs to be ON explcitly

**Review by andimarek — APPROVED** [review]
